e2e/helpers/wallet.helper.ts

```diff
@@ -0,0 +1,138 @@
+import { mnemonicToSeedSync } from 'bip39';
+import { BIP32Interface, fromSeed } from 'bip32';
+import * as ecc from 'tiny-secp256k1';
+import {
+    initEccLib,
+    payments,
+    Psbt,
+    networks,
+    Payment,
+    Transaction,
+} from 'bitcoinjs-lib';
+import { convertToSatoshi } from '@e2e/helpers/common.helper';
+
+initEccLib(ecc);
+
+export class WalletHelper {
+    private mnemonic: string;
+    private seed: Buffer;
+    private root: BIP32Interface;
+
+    constructor(
+        mnemonic = 'select approve zebra athlete happy whisper parrot will yellow fortune demand father',
+    ) {
```

Comment:
any reason for providing the mnemonic? Why generate a random seed every time?

Reply:
Using a fixed mnemonic ensures that the generated addresses and keys are consistent across different runs.
The seed is generated using the `mnemonicToSeedSync` function from the bip39 library, which takes the mnemonic as input. Since the mnemonic is fixed the seed generated will always be the same for default mnemonic.

Reply:
Is there a need to have consistent addresses across different runs?

Reply:
yup, predictable outputs are essential for verifying the correctness of the implementation

Reply:
Can you point me to the part of the code where we are using determinism in the tests? I don't see any hardcoded addresses or anything

Reply:
The `generateAddresses` method generates addresses based on the deterministic seed and root key.

Reply:
I ran the tests with a random seed phrase and all the tests passed. This confirms my hypothesis that
1. we aren't using the determinism in tests and
2. we don't need hardcoded seed phrases

Therefore, please update the tests to use a random seed.

Reply:
@theanmolsharma  The initial idea was to have to have a precomputed scan tweak  hence the determinism, but with the current approach, i think we do not such determinism anymore.  @notTanveer , kindly go ahead and use a random seed phrase.

Reply:
I also remember talking about this in one of the sync calls. Precomputed scan tweaks are not possible.